### ISSUE TITLE
[HOLD for payment 2023-03-09] Keyboard is visible for a brief moment on request/send money page after getting back from currency search page

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#15085]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Open the app in android
2. Open any report
3. Click on plus
4. Click on Request money/ send money/ split bill
5. Click on currency
6. On currency search page, when keyboard is displayed for search, click on 'X' to close and get back to previous page

## Expected Result:
Keyboard should close first and then app should get back to previous request/send money screen

## Actual Result:
App navigates to request/send money screen first and then closes the keyboard (works properly on mWeb android chrome)

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native

**Version Number:** 1.2.70-1
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/218354150-0006248d-a0c9-4717-b341-8147d7320f82.mp4

https://user-images.githubusercontent.com/43996225/218354181-6d0eeffe-3009-4383-ade2-88fe0749b6b9.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @dhanashree-sawant
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1676043921470679

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @mateocole (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by narefyev91** [comment]
Hello, I'm Nicolay from Callstack and I'm interested in taking and analysing this issue to work on a fix for it.

**Comment by narefyev91** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
App doesn't automatically hide keyboard on android - while moving between screens.

### What is the root cause of that problem?
The main issue - when user goes back - no events fires to app to dismiss keyboard (if keyboard has been already opened)

### What changes do you think we should make in order to solve the problem?
The main change should be:
1) using new [event handler](https://reactnavigation.org/docs/navigation-events/#beforeremove) to detect when user wants to live current screen.
2) when event triggers - needs to close keyboard (if keyboard has been already opened)
Example:
```
        this.beforeRemoveSubscription = this.props.navigation.addListener('beforeRemove', () => {
            if (this.props.isKeyboardShown) {
                Keyboard.dismiss();
            }
        });
```

### What alternative solutions did you explore?
We can solve issue in 2 ways - globally and locally. 
For **global one** - we can add subscription to event `beforeremove` inside `ScreenWrapper` class and add in compose `withKeyboardState` HOC - which will guaranty that keyboard will be closed on any screen which was wrapped in  `ScreenWrapper`
For **local one** - doing the same steps - but only under `IOUCurrencySelection` class

**Comment by MelvinBot** [comment]
Looks like something related to `react-navigation` may have been mentioned in this issue discussion.

As a reminder, please make sure that all proposals are not workarounds and that any and all attempt to fix the issue holistically have been made before proceeding with a solution. Proposals to change our `DeprecatedCustomActions.js` files should not be accepted.

Feel free to drop a note in [#expensify-open-source](https://expensify.slack.com/archives/C01GTK53T8Q) with any questions.

**Comment by mountiny** [comment]
We are trying to avoid making changes related to react-navigation now as we are reworking this. I think we could try this solution locally @narefyev91 and confirm it works well there on all the native platforms.

Can you please create a PR and fill in the template with testing done on all the platforms? Thank you very much!

**Comment by MelvinBot** [comment]
@narefyev91, @mateocole, @mountiny Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by mountiny** [comment]
PR is being reviewed.

**Comment by mountiny** [comment]
PR is going to be merged soon

**Comment by mountiny** [comment]
Merged

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.2.77-4 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/15205


If no regressions arise, payment will be issued on 2023-03-09. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@mountiny] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@mountiny] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@mountiny] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@narefyev91 / @0xmiroslav] Propose [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@mateocole] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by mateocole** [comment]
Okay, here is an upwork post for everyone to apply to for payment:
https://www.upwork.com/jobs/~01115b5c443336b31a

**Comment by mountiny** [comment]
This is only going to be:
- 250 for @dhanashree-sawant
- 1000 for @0xmiroslav 

@narefyev91 is handled differently

**Comment by dhanashree-sawant** [comment]
Hi @mountiny, thanks I have applied to the upwork job.

**Comment by mateocole** [comment]
Okay contracts sent!

**Comment by mateocole** [comment]
Alright payments sent

**Comment by syedsaroshfarrukhdot** [comment]
I believe i reported a similar issue a few months ago and we ended up resolving a dupe issue while original issue was on hold https://github.com/Expensify/App/issues/13449. So i think i am also eligible for reporting bonus here.

**Comment by mountiny** [comment]
Ltts keep it in that issue, even on hold I the reporting bonus should be paid I believe once we will put it off hold

**Comment by JmillsExpensify** [comment]
Jumping in here. Doesn't the fix for this issue also fix https://github.com/Expensify/App/issues/13449? In fact, I've tested that issue and I'm no longer able to reproduce it, so I'm pretty certain the answer is yes.

**Comment by JmillsExpensify** [comment]
Assuming we all agree on that last comment, then I agree that @syedsaroshfarrukhdot should be eligible for a reporting bonus and we should close the related issue. I'm happy to do it there, though let's first ensure that we agree on the next steps in this issue before I move forward.

**Comment by mountiny** [comment]
Checking, seems like this is the case. I think we can close and pay @syedsaroshfarrukhdot
